### ISSUE TITLE
A100 80G batch_size=4 OOM

### ISSUE THREAD

**Issue by minushuang** [ZhengPeng7/BiRefNet#83]
Hi, I noticed that you trained the model on 8 * A100-80G GPUS with batchsize=4 , and I followed the training settings as you mentioned in the paper, which means  I set the input_size=1024x1024, bs=4 on 8 * A100-80G GPUS too. 

but when i run train.sh, it occours a "OOM" error, with the cmd  `watch -n 0.1 nvidia-smi` when trainning, it shows that, NO.0 GPU oom, and the others is ok, the NO.0 GPU has more than one process on it like bellow.
![image](https://github.com/user-attachments/assets/04e44172-5007-4053-b874-8eded5a4449e)

**Comment by ZhengPeng7** [comment]
Really? Let me check it now. I'm lucky to just finish a training and have those idle GPUs.

**Comment by ZhengPeng7** [comment]
It looks good on my machine with the default settings (compile=False now, but True should also be okay). Maybe you can check the env?
![截屏2024-09-13 21 28 00](https://github.com/user-attachments/assets/aa01a519-d420-4a01-bad2-e5191cdbbd06)

**Comment by minushuang** [comment]
@ZhengPeng7  thank u for your reply. 

It's strange, my environment is configured exactly according to the readme, I'll double check it later.  

One more question, it's about input image size, considering that the size of the images in my usage scenario (including the training set) are all 1340x1785 (wxh), with an aspect ratio of 0.75, I changed the input size from 1024x1024 to 768x1024.  

This not only better suits my usage scenario, but also avoids OOM error when set batchsize=4. 

However, I'm not sure about how much this size change will affect the results, whether positively or negatively.

Could you offer some advice on this?

thank u very much.

**Comment by ZhengPeng7** [comment]
Sure. It should work directly after you change the resolution to the new one. But make sure you're using the latest version, as several changes were made to the adaptable resolution for training, about one week ago.

**Comment by ZhengPeng7** [comment]
It should work better in your cases. 1024x1024 is a trade-off value for all cases in practical using.

**Comment by minushuang** [comment]
> It should work better in your cases. 1024x1024 is a trade-off value for all cases in practical using.

OK, thans again, that's what i want.

**Comment by minushuang** [comment]
> Sure. It should work directly after you change the resolution to the new one. But make sure you're using the latest version, as several changes were made to the adaptable resolution for training, about one week ago.

Yes, It can work normally after chaning the config.input_size from 1024x1024 to 768x1024.

During the training process, I found that,the 8 GPUs are outputting different loss values.  Seven of them are relatively close, but one consistently has a much higher loss than the other seven, when under one same interation.

Is this normal? What do you think could be causing this issue?

![image](https://github.com/user-attachments/assets/b0f46fe3-dc61-4d80-bb4c-d4b5e45c0a42)

**Comment by ZhengPeng7** [comment]
In my training log, two or three of the eight values are larger than others. I think this is common and not a problem.

**Comment by minushuang** [comment]
> In my training log, two or three of the eight values are larger than others. I think this is common and not a problem.

Okay, many thanks for your patient reply.
